### PR TITLE
chore(github): add pull request template

### PR DESCRIPTION
**Pull request by rwaskiewicz**
Meta: This is the current shape I have for the PR template. It would apply to both the core stencil team and external contributors, as GH does not have a way to differentiate the two ATM.  LMK what yinz think
<hr/>

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Unit tests (`npm test`) were run locally and passed
- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [ ] Prettier (`npm run format`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Adds a template to GH for pull requests. All new PR's will be shown this and folks will expected to fill this out

## What is the current behavior?

There is no template, which allows for a greater spread of quality in PR summaries we get

GitHub Issue Number: N/A

## What is the new behavior?

- This template is shown on every new PR

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

- N/A, I expect GH to pick this up automatically once it lands

## Other information

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Pull request checklist
+
+Please check if your PR fulfills the following requirements:
+- [ ] Tests for the changes have been added (for bug fixes / features)
+- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
+- [ ] Build (`npm run build`) was run locally and any changes were pushed
+- [ ] Unit tests (`npm test`) were run locally and passed
+- [ ] E2E Tests (`npm run test.karma.prod`) were run locally and passed
+- [ ] Prettier (`npm run format`) was run locally and passed
```
Comment (ltm) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Prettier is currently `npm run prettier` (but I would not be opposed to making it `npm run format`):
https://github.com/ionic-team/stencil/blob/252415a5cc94cad9ea0a5eddb73bae32d1845c96/package.json#L39

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Pull request type
+
+<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 
+
+<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 
+
+Please check the type of change your PR introduces:
```
Comment (ltm) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: A side effect of using a task list here is that all PRs will have "8 of 14 tasks" completed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (rwaskiewicz) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: Bah, that's my fault. In a former life I named the script 'format'. Let me change that realllll quick

**Review by splitinfinities — APPROVED** [review]
This is huge! Thank you so much for doing this!!

**Pull request merged**